save-demo/src/main/kotlin/com/saveourtool/save/demo/utils/KubernetesUtils.kt

```diff
@@ -0,0 +1,139 @@
+    val job = Job().apply {
+        metadata = ObjectMeta().apply {
+            name = jobNameForDemo(demo)
+        }
+        spec = JobSpec().apply {
+            parallelism = REPLICAS_PER_DEMO
+            ttlSecondsAfterFinished = TTL_AFTER_COMPLETED
+            backoffLimit = 0
+            template = PodTemplateSpec().apply {
+                spec = PodSpec().apply {
+                    if (kubernetesSettings.useGvisor) {
+                        nodeSelector = mapOf(
+                            "gvisor" to "enabled"
+                        )
+                        runtimeClassName = "gvisor"
+                    }
+                    metadata = ObjectMeta().apply {
+                        labels = mapOf(
+                            DEMO_ORG_NAME to demo.organizationName,
+                            DEMO_PROJ_NAME to demo.projectName,
+                            DEMO_VERSION to "manual",
+                            // "baseImageName" to baseImageName
+                            "io.kompose.service" to "save-demo-agent",
+                        )
+                    }
+                    // If agent fails, we should handle it manually (update statuses, attempt restart etc.)
+                    restartPolicy = "Never"
+                    containers = listOf(demoAgentContainerSpec(demo.sdk.toSdk().baseImageName(), kubernetesSettings))
+                }
+            }
+        }
+    }
+    logger.debug { "Attempt to create Job from the following spec: $job" }
+    try {
+        resource(job).create()
+        with(demo) {
+            logger.info("Created Job for demo $organizationName/$projectName")
+        }
+    } catch (kex: KubernetesClientException) {
+        with(demo) {
+            throw KubernetesRunnerException("Unable to create a job for demo $organizationName/$projectName", kex)
+        }
+    }
+}
+
+/**
+ * @param demo demo entity
+ * @return [ScalableResource] of [Job]
+ */
+fun KubernetesClient.getJobByName(demo: Demo): ScalableResource<Job> = batch()
+    .v1()
+    .jobs()
+    .inNamespace(namespace)
+    .withName(jobNameForDemo(demo))
+
+/**
+ * @param demo demo entity
+ * @return true if the resource is ready or exists (if no readiness check exists), false otherwise.
+ */
+fun KubernetesClient.isJobReady(demo: Demo) = getJobByName(demo).isReady
+
+/**
+ * @param demo demo entity
+ * @return list of ips of pods that are run by job associated with [demo]
+ */
+fun KubernetesClient.getJobPodsIps(demo: Demo) = getJobPods(demo)
+    .map { it.status.podIP }
+
+/**
+ * @param demo demo entity
+ * @return list of pods that are run by job associated with [demo]
+ */
+fun KubernetesClient.getJobPods(demo: Demo): List<Pod> = pods()
+    .withLabel(DEMO_ORG_NAME, demo.organizationName)
+    .withLabel(DEMO_PROJ_NAME, demo.projectName)
+    .list()
+    .items
+
+// fun <T> Filterable<T>.withDemoLabels(demo: Demo): Filterable<T> = withLabel(DEMO_ORG_NAME, demo.organizationName)
```

Comment:
commented code?

Reply:
Done

Reply:
I think we need finally to make diktat not add spaces after such comments (where they contain code)